File: web_app/business/GameController.py
"""
GameController.py
"""
from web_app.business.AccessQuestions import AccessQuestions

class GameController(object):
    """Controls the game"""

    instance = None

    def __init__(self, max_questions):
        self.max_questions = max_questions
        self.access_questions = AccessQuestions()
        self.questions = None
        self.curr_question = None
        self.question_count = -1
        self.score = -1
        self.is_started = False

    @classmethod
    def get_instance(cls, max_questions):
        """Fetch the one and only instance"""
        if GameController.instance is None:
            GameController.instance = GameController(max_questions)

        return GameController.instance

    @classmethod
    def destroy(cls):
        """Destroy the single instance"""
        GameController.instance = None

    def start(self):
        """Start the game by grabing a set of questions"""
        self.question_count = 0
        self.score = 0
        self.questions =\
        self.access_questions.get_random_questions(self.max_questions)
        self.is_started = True

    def get_next_question(self):
        """Return the next question"""

        if self.question_count == self.max_questions:
            self.curr_question = None
        else:
            self.curr_question = self.questions[self.question_count]
            self.question_count += 1

        return self.curr_question

    def increase_score(self):
        self.score += 1

    def evaluate_answer(self, answer_index):
        """Check if the current question answer equals answer_index"""
        result = True if answer_index == self.curr_question.answer else False
        return result

    def is_finished(self):
        """Check if the game is finished"""
        return self.max_questions == self.question_count

File: TriviaSmack/app/src/main/java/comp4350/triviasmack/business/GameController.java
package comp4350.triviasmack.business;

import java.util.ArrayList;

import comp4350.triviasmack.application.Services;
import comp4350.triviasmack.objects.Question;

public class GameController {

    private static GameController instance = null;
    private final static int maxQuestions = 2;

    private ArrayList<Question> questions;
    private int questionCount;
    private Question currQuestion;
    private int score;
    private boolean started;
    private ServerAccess serverAccess;

    protected GameController(){
        questions = null;
        questionCount = -1;
        currQuestion = null;
        score = -1;
        serverAccess = Services.getServerAccess();
    }

    public static GameController getInstance() {
        if(instance == null) {
            instance = new GameController();
        }
        return instance;
    }

    private ArrayList<Question> getQuestions(){
        return serverAccess.getRandomQuestions();
    }

    public int getScore(){ return score; }

    public void start(){
        questionCount = 0;
        score = 0;
        questions = getQuestions();
        started = true;
    }

    public Question getNextQuestion(){
        if (questionCount == maxQuestions) {
            currQuestion = null;
        }
        else {
            currQuestion = questions.get(questionCount);
            questionCount++;
        }

        return currQuestion;
    }

    public boolean evaluateAnswer(String playersAnswer){
        boolean result = false;
        String answer = currQuestion.options[currQuestion.answer];

        if (playersAnswer.equalsIgnoreCase(answer)){
            result = true;
        }
        return result;
    }

    public void increaseScore(){ score++; }

    public boolean isStarted(){ return started; }

    public boolean finished(){
        return maxQuestions == questionCount;
    }

    public static void destroy(){ instance = null; }
}
